### PR TITLE
docs: update ApisixRoute/v2 reference

### PR DESCRIPTION
**Pull request by navendu-pottekkat**
Signed-off-by: Navendu Pottekkat <navendu@apache.org>

Updates the documentation of ApisixRoute/v2 reference page.

Similar changes would be made across the reference section after approving this PR.

### CODE REVIEW

**Comment by navendu-pottekkat** [comment]
Any idea why there is this lint error? https://github.com/apache/apisix-ingress-controller/actions/runs/3367934841/jobs/5585981373#step:5:6

It is not obvious what the issue is to me.

**Comment by juzhiyuan** [comment]
image

Error from other docs, I think we can ignore them in this PR.

File: docs/en/latest/references/apisix_route_v2.md [hunk 1]
```diff
@@ -21,83 +26,85 @@ title: ApisixRoute/v2 Reference
...
+| http[].name                          | string (required)  | Route rule name.                                                                                                                                                                          |
+| http[].priority                      | integer            | Route priority. Used to determined which Route to use when multiple routes contain the same URI. Large number means higher priority.                                                      |
+| http[].timeout                       | object             | Timeouts for connecting, sending, and receiving messages between Ingress and the service. This will overwrite the value configured in `ApisixUpstream` resource.                          |
+| http[].timeout.connect               | string             | Connect timeout in the form "72h3m0.5s".                                                                                                                                                  |
```
Comment (juzhiyuan) [inline docs/en/latest/references/apisix_route_v2.md:43]: Hi @navendu-pottekkat, is this expected?

image

**Review by juzhiyuan — COMMENTED** [review]
Other content LGTM

File: docs/en/latest/references/apisix_route_v2.md (hunk 1 above)
Comment (navendu-pottekkat) [reply docs/en/latest/references/apisix_route_v2.md:43]: @juzhiyuan I did not understand your question.

Comment (juzhiyuan) [reply docs/en/latest/references/apisix_route_v2.md:43]: image

This form is valid?

Comment (navendu-pottekkat) [reply docs/en/latest/references/apisix_route_v2.md:43]: I don't know. I did not change it. @tao12345666333 could you confirm?

**Review by juzhiyuan — APPROVED** [review]
Other LGTM

File: docs/en/latest/references/apisix_route_v2.md [hunk 2]
```diff
@@ -77,11 +77,11 @@ The table below describes each of the attributes in the spec. The fields `apiVer
...
-| stream[].backend.servicePort         | integer or string  | Port number or the name defined in the service object of the backend (depricated).                                                                                                        |
-| stream[].backend.resolveGranularity  | string             | See [Service resolution granularity](#service-resolution-granularity) for details (depricated).                                                                                           |
-| stream[].backend.subset              | string             | Subset for the target service (depricated). Should be pre-defined in the `ApisixUpstream` resource.                                                                                       |
+| stream[].backend                     | object             | Backend service (depreciated). Use `http[].backends` instead.                                                                                                                              |
```
Comment (juzhiyuan) [inline docs/en/latest/references/apisix_route_v2.md:80]: Hi, I think `deprecated` is correct 🤔

Comment (navendu-pottekkat) [reply docs/en/latest/references/apisix_route_v2.md]: Yes. I thought I was wrong and changed it... Deprecated is the correct one. I have checked.

I have reverted the commit

File: docs/en/latest/references/apisix_route_v2.md (hunk 1 above)
Comment (tao12345666333) [reply docs/en/latest/references/apisix_route_v2.md:43]: let me check it.

File: docs/en/latest/references/apisix_route_v2.md [hunk 3]
```diff
@@ -21,83 +26,85 @@ title: ApisixRoute/v2 Reference
...
+| stream[].match.ingressPort           | integer (required) | Listening port in the Ingress proxy server. This port should be defined in the [APISIX configuration](https://github.com/apache/apisix/blob/master/conf/config-default.yaml#L101).        |
+| stream[].backend                     | object             | Backend service (deprecated). Use `http[].backends` instead.                                                                                                                              |
+| stream[].backend.serviceName         | string             | Name of the backend service (depricated). The service and the `ApisixRoute` resource should be created in the same namespace.                                                             |
+| stream[].backend.servicePort         | integer or string  | Port number or the name defined in the service object of the backend (depricated).                                                                                                        |
```
Comment (juzhiyuan) [inline docs/en/latest/references/apisix_route_v2.md:82]:
```suggestion
| stream[].backend.servicePort         | integer or string  | Port number or the name defined in the service object of the backend (deprecated).                                                                                                        |
```

**Review by juzhiyuan — COMMENTED** [review]
Other LGTM

**Review by juzhiyuan — APPROVED** [review]

**Comment by navendu-pottekkat** [comment]
I have fixed the link check error in a different PR. All checks will pass here.

**Comment by navendu-pottekkat** [comment]
@juzhiyuan @tao12345666333 I have updated the v2beta3 docs as well in this PR. Please review and merge.

**Review by lingsamuel — APPROVED** [review]

**Pull request merged**